rspace/src/main/scala/coop/rchain/rspace/history/package.scala

```diff
@@ -0,0 +1,152 @@
+package coop.rchain.rspace
```

Comment:
Please review this carefully, I spend a lot of time here and encountered some bugs. I'm pretty confident all is good, but this is crucial point. We have in codebase tuple space hash for empty trie hardcoded, so this should serve as a protection for encoding corruption. 
https://github.com/rchain/rchain/blob/1724688c15cda74dad2c8dd26302ea6a7a880334/casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala#L218-L222
Might be there should be a test for that, because encoding tests did not catch bug that I had in this code - only by empty trie hash mismatch I figured out whats wrong.

Reply:
Please create an issue to write these tests and review it in detail.